ui/src/alerting/components/notifications/NewRuleOverlay.reducer.ts

```diff
@@ -0,0 +1,88 @@
+// Libraries
+import {v4} from 'uuid'
+
+// Types
+import {NotificationRuleBox, StatusRuleItem, TagRuleItem} from 'src/types'
+
+export type RuleState = NotificationRuleBox
```

Comment:
I'm confused about this name. I know I originally suggested `IDBox`, but now am thinking that doesn't make sense either. The conventional definition of a boxed type is a type that wraps a value with an object to force it to be allocated on the heap. This doesn't wrap an object and it is already allocated on the heap. The only other `Box` type in the UI right now is the `CancelBox`, which does wrap a value but not for the purposes of heap allocation. ¯\_(ツ)_/¯

I do think it would be really nice to have a convention for types that represent our working browser-side copy of some resource that exists on the server. I've seen us use the `Working` prefix for that before, e.g. `WorkingView`. I could also imagine us using `Draft` or `Client`, e.g. `DraftNotificationRule`, `DraftView`, ...

Probably doesn't matter much, but nice names are nice :)

Reply:
I agree.  This was my least favorite part of this PR.  I like `Draft` as a namespace.